src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -142,29 +142,46 @@ internal double ConvertMoerToGramsPerKilowattHour(double value)
         return value * LBS_TO_GRAMS_CONVERSION_FACTOR / MWH_TO_KWH_CONVERSION_FACTOR;
     }
 
-    private IEnumerable<EmissionsData> ConvertToEmissionsData(IEnumerable<GridEmissionDataPoint> data)
+    private IEnumerable<EmissionsData> ConvertToEmissionsData(IEnumerable<GridEmissionDataPoint> gridEmissionDataPoints)
     {
+        var defaultDuration = GetDurationFromGridEmissionDataPointsOrDefault(gridEmissionDataPoints, TimeSpan.Zero);
+        
         // Linq statement to convert WattTime forecast data into EmissionsData for the CarbonAware SDK.
-        return data.Select(e => new EmissionsData() 
+        return gridEmissionDataPoints.Select(e => new EmissionsData() 
                     { 
                         Location = e.BalancingAuthorityAbbreviation, 
                         Rating = ConvertMoerToGramsPerKilowattHour(e.Value), 
                         Time = e.PointTime,
-                        Duration = FrequencyToTimeSpan(e.Frequency)
+                        Duration = FrequencyToTimeSpanOrDefault(e.Frequency, defaultDuration)
                     });
     }
 
-    private TimeSpan GetDurationFromGridEmissionDataPoints(GridEmissionDataPoint? firstPoint, GridEmissionDataPoint? secondPoint)
+    private TimeSpan GetDurationFromGridEmissionDataPoints(IEnumerable<GridEmissionDataPoint> gridEmissionDataPoints)
     {
+        var firstPoint = gridEmissionDataPoints.FirstOrDefault(); 
+        var secondPoint = gridEmissionDataPoints.Skip(1)?.FirstOrDefault();
+
         var first = firstPoint ?? throw new WattTimeClientException("Too few data points returned"); 
         var second = secondPoint ?? throw new WattTimeClientException("Too few data points returned");
 
-        return second.PointTime - first.PointTime;
+        return (first.PointTime - second.PointTime).Duration();
```

Comment:
Why do we do `.Duration()` now instead of how it was?

Reply:
<.Duration()> gives an absolute value of the given Timespan object and thats what we wanted

Reply:
Because WattTime data can be chronological OR reverse chronological, `.Duration()` gives us the absolute value.

Reply:
It might be more stylistically accurate and easier to read if you used the TimeSpan.Subtract method directly (and omitted the .Duration)

Reply:
@gfmatthews Agreed on `TimeSpan.Subtract`.  But we still need to use `Duration` to handle getting the absolute value.  I've added a comment to that effect

Reply:
`Subtract` method could return a negative value and we would still need to use Duration() to convert it.

Reply:
Right but if we get a negative value out of subtract, that's an indication that something else has gone wrong isn't it?

Reply:
@gfmatthews In this case, no.  WattTime will return data in chronological order as well as reverse-chronological order but we only care about the distance between the points, not the direction.

Chronological: `2022-08-01T12:00:00Z` - `2022-08-01T12:05:00Z` = `- 00:05:00`  (negative)
Reverse-Chronological: `2022-08-01T12:05:00Z` - `2022-08-01T12:00:00Z` = `00:05:00` (positive)

Both are valid for estimating the positive `frequency` or `duration` of the individual data points.